thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detection/algorithm/stage/AnomalyDetectionStage.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.thirdeye.detection.algorithm.stage;
+
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import com.linkedin.thirdeye.detection.DataProvider;
+import java.util.List;
+
+
+/**
+ * The anomaly detection stage
+ */
+public interface AnomalyDetectionStage extends BaseDetectionStage {
+  /**
+   * Run detection in the specified time range and return a list of anomalies
+   * @param startTime start time stamp
+   * @param endTime end time stamp
+   * @param provider centralized data source for time series, anomalies, events, etc.
+   * @return list of anomalies
+   */
+  List<MergedAnomalyResultDTO> runDetection(long startTime, long endTime, DataProvider provider);
```

Comment:
is the interface an algorithm has to implement? why are we providing access to the dataProvider to the underlying implementation by default? Shouldn't the framework do the work to fetch the data and invoke the algorithm with the data?

Reply:
We can have a higher level API where the user only needs to define what data it needs to fetch and the framework will fetch the data and then run the stage with the data.